pkg/apiaccess/types.go

```diff
@@ -0,0 +1,245 @@
+	// Criteria by which to narrow the scope of keys to be returned.
+	Scope APIAccessKeySearchScope `json:"scope"`
+	// A list of key types to be included in the search. If no types are provided, all types will be returned by default.
+	Types []APIAccessKeyType `json:"types"`
+}
+
+// APIAccessKeySearchResult - A list of all keys scoped to the current actor.
+type APIAccessKeySearchResult struct {
+	// A list of all keys scoped to the current actor.
+	Keys []APIAccessKey `json:"keys"`
+	// The next cursor, used for pagination. If a cursor is present, it means more keys can be fetched.
+	NextCursor string `json:"nextCursor"`
+}
+
+// APIAccessKeySearchScope - The scope of keys to be returned. Note that some filters only apply to certain key types.
+type APIAccessKeySearchScope struct {
+	// A list of key account IDs.
+	AccountIDs []int `json:"accountIds"`
+	// The ingest type of the key. Only applies to ingest keys, and does not affect user key filtering.
+	IngestTypes []APIAccessIngestKeyType `json:"ingestTypes"`
+	// A list of key user ids. Only applies to user keys, and does not affect ingest key filtering.
+	UserIDs []int `json:"userIds"`
+}
+
+// APIAccessUpdateIngestKeyInput - The `id` and data to update one or more keys.
+type APIAccessUpdateIngestKeyInput struct {
+	// The `id` of the key you want to update.
+	KeyID string `json:"keyId"`
+	// The name you want to assign to the key.
+	Name string `json:"name"`
+	// The notes you want to assign to the key.
+	Notes string `json:"notes"`
+}
+
+// APIAccessUpdateInput - The `id` and data to update one or more keys.
+type APIAccessUpdateInput struct {
+	// A list of the configurations of each ingest key you want to update.
+	Ingest []APIAccessUpdateIngestKeyInput `json:"ingest"`
+	// A list of the configurations of each user key you want to update.
+	User []APIAccessUpdateUserKeyInput `json:"user"`
+}
+
+// APIAccessUpdateKeyResponse - The response of the update keys mutation.
+type APIAccessUpdateKeyResponse struct {
+	// Lists all errors for keys that could not be updated. Each error maps to a single key input.
+	Errors []APIAccessKeyError `json:"errors"`
+	// Lists all successfully updated keys.
+	UpdatedKeys []APIAccessKey `json:"updatedKeys"`
+}
+
+// APIAccessUpdateUserKeyInput - The `id` and data to update one or more keys.
+type APIAccessUpdateUserKeyInput struct {
+	// The `id` of the key you want to update.
+	KeyID string `json:"keyId"`
+	// The name you want to assign to the key.
+	Name string `json:"name"`
+	// The notes you want to assign to the key.
+	Notes string `json:"notes"`
+}
+
+// ID - The `ID` scalar type represents a unique identifier, often used to
+// refetch an object or as key for a cache. The ID type appears in a JSON
+// response as a String; however, it is not intended to be human-readable.
+// When expected as an input type, any string (such as `"4"`) or integer
+// (such as `4`) input value will be accepted as an ID.
+type ID string
```

Comment:
This probably isn't needed.  I think the alerts package for the tutone config has an example for how to ignore it.

Reply:
Cool, I'll take a look. I was following the previously added package I think.

Reply:
I removed this and updated the tutone config for this package so it doesn't get created :)